app/src/main/java/com/hedvig/app/authenticate/insurely/InsurelyAuthViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.hedvig.app.authenticate.insurely
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class InsurelyAuthViewModel(
+    private val reference: String,
+    private val getDataCollectionUseCase: GetDataCollectionUseCase
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(ViewState())
+    val viewState: StateFlow<ViewState>
+        get() = _viewState
```

Comment:
We usually write this as
```kotlin
val viewState = _viewState.asStateFlow()
```
Infers the type well by itself too, pretty neat.
And might be good to keep it consistent in case we want to search for all such occasions in the project for whatever reason.